code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -335,3 +416,61 @@
 		. += "-HAIR_HIDDEN"
 
 	return .
+
+/obj/item/bodypart/r_leg/generate_icon_key()
+	RETURN_TYPE(/list)
+	. = ..()
+	if(left_leg_mask_key) // We do this so we can cache the versions with and without a mask, for when there's no left leg.
+		. += "-[left_leg_mask_key]"
+
+	return .
+
+/**
+ * This proc serves as a way to ensure that right legs don't overlap above left legs when their dir is WEST on a mob.
+ *
+ * It's using the `left_leg_mask_cache` to avoid generating a new mask when unnecessary, which means that there needs to be one
+ * for the proc to return anything.
+ *
+ * Arguments:
+ * * right_leg_icon_file - The icon file of the right leg overlay we're trying to apply a mask to.
+ * * right_leg_icon_state - The icon_state of the right leg overlay we're trying to apply a mask to.
+ * * image_dir - The direction applied to the icon, only meant for when the leg is dropped, so it remains
+ * facing SOUTH all the time.
+ *
+ * Returns the `/image` of the right leg that was masked, or `null` if the mask didn't exist.
+ */
+/obj/item/bodypart/r_leg/proc/generate_masked_right_leg(right_leg_icon_file, right_leg_icon_state, image_dir)
+	RETURN_TYPE(/image)
+	if(!left_leg_mask_key || !left_leg_mask_cache[left_leg_mask_key] || !right_leg_icon_file || !right_leg_icon_state)
+		return
+
+	var/icon/right_leg_icon = icon(right_leg_icon_file, right_leg_icon_state)
+	right_leg_icon.Blend(left_leg_mask_cache[left_leg_mask_key], ICON_MULTIPLY)
+	return image(right_leg_icon, right_leg_icon_state, layer = -BODYPARTS_LAYER, dir = image_dir)
```

Comment:
This hurts

Reply:
It does, but from ninja's own words, making an image out of an icon is actually very cheap, something something direct reference to rsc. I'd need to ask him again to be certain, though.